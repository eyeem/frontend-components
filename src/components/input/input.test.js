import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Input from './index';

const TEST_PLACEHOLDER = 'test placeholde';
const TEST_VALUE = 'test value';
const TEST_NAME = 'test name';

const testInputProps = {
  required: true,
  value: TEST_VALUE,
  onChange: () => onInputChangedHandler,
  name: TEST_NAME,
  placeholder: TEST_PLACEHOLDER
};

const onInputChangedHandler = () => 'changed';

afterEach(cleanup);

test('it renders default input', () => {
  const { container } = render(<Input inputProps />);
  const component = container.firstChild;
  expect(component).toBeTruthy();
  expect(component.firstChild).toHaveProperty('type');
  expect(component.firstChild.type).toBe('text');
  expect(component).toBeTruthy();
  expect(component).toMatchSnapshot();
});

test('it accepts inputProps', () => {
  const { container } = render(<Input inputProps={testInputProps} />);
  const component = container.firstChild;
  const inputElement = component.firstChild;
  expect(inputElement).toHaveProperty('placeholder');
  expect(inputElement.placeholder).toBe(TEST_PLACEHOLDER);
  expect(inputElement).toHaveProperty('name');
  expect(inputElement.name).toBe(TEST_NAME);
  expect(inputElement).toHaveProperty('required');
  expect(inputElement).toHaveProperty('type');
  expect(inputElement.type).toBe('text');
  expect(inputElement).toHaveProperty('value');
  expect(inputElement.value).toBe(TEST_VALUE);
  expect(component).toMatchSnapshot();
});
