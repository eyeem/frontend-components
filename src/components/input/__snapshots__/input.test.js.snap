// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it accepts inputProps 1`] = `
.c0 {
  position: relative;
  margin-top: 16px;
}

.c0:first-child {
  margin-top: 0;
}

.c1 {
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
  font-size: 16px;
  line-height: 20px;
  padding: 9px 12px 10px 12px;
  max-height: 40px;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-radius: 2px;
  box-shadow: none;
  color: #000000;
  border-color: #cccccc;
  background-color: #ffffff;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: rgba(0,0,0,.4);
}

.c1::-moz-placeholder {
  color: rgba(0,0,0,.4);
}

.c1:-ms-input-placeholder {
  color: rgba(0,0,0,.4);
}

.c1::placeholder {
  color: rgba(0,0,0,.4);
}

.c1:focus {
  border-color: #22cc88;
}

.c1:focus::-webkit-input-placeholder {
  color: rgba(0,0,0,.2);
}

.c1:focus::-moz-placeholder {
  color: rgba(0,0,0,.2);
}

.c1:focus:-ms-input-placeholder {
  color: rgba(0,0,0,.2);
}

.c1:focus::placeholder {
  color: rgba(0,0,0,.2);
}

<div
  class="c0"
>
  <input
    class="c1"
    name="test name"
    placeholder="test placeholde"
    required=""
    type="text"
    value="test value"
  />
</div>
`;

exports[`it renders default input 1`] = `
.c0 {
  position: relative;
  margin-top: 16px;
}

.c0:first-child {
  margin-top: 0;
}

.c1 {
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
  font-size: 16px;
  line-height: 20px;
  padding: 9px 12px 10px 12px;
  max-height: 40px;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-radius: 2px;
  box-shadow: none;
  color: #000000;
  border-color: #cccccc;
  background-color: #ffffff;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: rgba(0,0,0,.4);
}

.c1::-moz-placeholder {
  color: rgba(0,0,0,.4);
}

.c1:-ms-input-placeholder {
  color: rgba(0,0,0,.4);
}

.c1::placeholder {
  color: rgba(0,0,0,.4);
}

.c1:focus {
  border-color: #22cc88;
}

.c1:focus::-webkit-input-placeholder {
  color: rgba(0,0,0,.2);
}

.c1:focus::-moz-placeholder {
  color: rgba(0,0,0,.2);
}

.c1:focus:-ms-input-placeholder {
  color: rgba(0,0,0,.2);
}

.c1:focus::placeholder {
  color: rgba(0,0,0,.2);
}

<div
  class="c0"
>
  <input
    class="c1"
    type="text"
  />
</div>
`;
